definitions/npm/react-router-dom_v6.x.x/flow_v0.201.x-/react-router-dom_v6.x.x.js

```diff
@@ -226,6 +226,245 @@ declare module 'react-router-dom' {
+      ' _hasFetcherDoneAnything '?: boolean,
+    },
+    Loading: {
+      state: 'loading',
+      formMethod: RemixFormMethod | V7_FormMethod | void,
+      formAction: string | void,
+      formEncType: FormEncType | void,
+      formData: FormData | void,
+      data: TData | void,
+      ' _hasFetcherDoneAnything '?: boolean,
+    },
+    Submitting: {
+      state: 'submitting',
+      formMethod: RemixFormMethod | V7_FormMethod,
+      formAction: string,
+      formEncType: FormEncType,
+      formData: FormData,
+      data: TData | void,
+      ' _hasFetcherDoneAnything '?: boolean,
+    },
+  };
+
+  declare export type Fetcher<TData = any> = FetcherStates<TData>[$Keys<
+    FetcherStates<TData>
+  >];
+
+  declare type BlockerBlocked = {|
+    state: 'blocked',
+    reset(): void,
+    proceed(): void,
+    location: Location,
+  |};
+
+  declare type BlockerUnblocked = {|
+    state: 'unblocked',
+    reset: void,
+    proceed: void,
+    location: void,
+  |};
+
+  declare type BlockerProceeding = {|
+    state: 'proceeding',
+    reset: void,
+    proceed: void,
+    location: Location,
+  |};
+
+  declare export type Blocker =
+    | BlockerUnblocked
+    | BlockerBlocked
+    | BlockerProceeding;
+
+  /**
+   * State maintained internally by the router.  During a navigation, all states
+   * reflect the the "old" location unless otherwise noted.
+   */
+  declare export type RouterState = {|
+    /**
+     * The action of the most recent navigation
+     */
+    historyAction: HistoryAction,
+
+    /**
+     * The current location reflected by the router
+     */
+    location: Location,
+
+    /**
+     * The current set of route matches
+     */
+    matches: AgnosticDataRouteMatch[],
+
+    /**
+     * Tracks whether we've completed our initial data load
+     */
+    initialized: boolean,
+
+    /**
+     * Current scroll position we should start at for a new view
+     *  - number -> scroll position to restore to
+     *  - false -> do not restore scroll at all (used during submissions)
+     *  - null -> don't have a saved position, scroll to hash or top of page
+     */
+    restoreScrollPosition: number | false | null,
+
+    /**
+     * Indicate whether this navigation should skip resetting the scroll position
+     * if we are unable to restore the scroll position
+     */
+    preventScrollReset: boolean,
+
+    /**
+     * Tracks the state of the current navigation
+     */
+    navigation: Navigation,
+
+    /**
+     * Tracks any in-progress revalidations
+     */
+    revalidation: RevalidationState,
+
+    /**
+     * Data from the loaders for the current matches
+     */
+    loaderData: RouteData,
+
+    /**
+     * Data from the action for the current matches
+     */
+    actionData: RouteData | null,
+
+    /**
+     * Errors caught from loaders for the current matches
+     */
+    errors: RouteData | null,
+
+    /**
+     * Map of current fetchers
+     */
+    fetchers: Map<string, Fetcher<>>,
```

Comment:
These `Map`s don't actually do what you might be expecting. I believe this usage of `Map` is looking for an instance of `Map` instead of object mappings.

Proper way in flow would be the following. Same for `blockers`

```suggestion
    fetchers: { [key: string]:  Fetcher<> },
```

And here's a try flow of what I expected Map to do but still work as expected
https://flow.org/try/#1N4Igxg9gdgZglgcxALlAJwKYEMwBcD6aArlLnALYYrgA2WAzvXGCADQgYAeOBARgJ74AJhhhYiNXClzEM7DFCLl602QF92kEdVz8ADhgAEAQUMBeQwFksegDz0ZcKAlaGHaJwgB8AHSh-IKAdDLGQTc0NgELCAchjDNTYQADcMNCZoamSABgA6ACZsgFZc7JA1IA

Reply:
ha right, thank you; I imported typescript almost as is and didn't pay attention to some important nuances. Sorry.

Reply:
That's what reviewers are for :) No harm done